### DISCUSSION TITLE
Preserve selection highlight when interacting with the toolbar

### DISCUSSION THREAD

**Discussion by noahnu** [facebook/lexical#6050] [Q&A] [answered]
Is it possible to keep the selected text highlighted while the user's focus changes to an input in the toolbar? For example, in the below video I highlight some text and then go to change the font size. When I start typing into the font size input, I no longer know which text has been selected until I make my change.

https://github.com/facebook/lexical/assets/1297096/c846ab4d-6687-4110-8fe4-9e2873fd4dd7

In Google docs, they just grey out unfocused selection: 
<img width="343" alt="image" src="https://github.com/facebook/lexical/assets/1297096/447d61cb-d0f5-4d96-8f8d-1e2d1dace456">

**Comment by Chigozie-Gerald** [comment]
Try wrapping your toolbar items in a button. I believe buttons are non-intrusive

**Reply by ashiq-prospero** [reply]
> Try wrapping your toolbar items in a button. I believe buttons are non-intrusive

it works for many cases but sometime we have to focus in input in color picker, font search etc in those case we lose focus from editor so in those case we need fake selection highlight

we have this, but this is implemented way to bad
```
import { SelectionAlwaysOnDisplay } from "@lexical/react/LexicalSelectionAlwaysOnDisplay";

<SelectionAlwaysOnDisplay />
```

**Comment by hurrii** [comment]
@trueadm @zurfyx is there a way to achieve that?

**Comment by ashiq-prospero** [comment]
Any plan to add props for SelectionAlwaysOnDisplay like we can customise the highlight background color and text color, or just pass custom rendered component in prop or just className which will allow us to customise the look?

```
import { SelectionAlwaysOnDisplay } from "@lexical/react/LexicalSelectionAlwaysOnDisplay";
```

## Some expectation
```
<SelectionAlwaysOnDisplay className="fake-selection" />
```
or
```
<SelectionAlwaysOnDisplay style={{color:'#555', backgroundColor:'#ddd'}} />
```

**Comment by ashiq-prospero** [comment, marked as answer]
> Is it possible to keep the selected text highlighted while the user's focus changes to an input in the toolbar? For example, in the below video I highlight some text and then go to change the font size. When I start typing into the font size input, I no longer know which text has been selected until I make my change.


you can use SelectionAlwaysOnDisplay as of now, it works but not so great.
```
import { SelectionAlwaysOnDisplay } from "@lexical/react/LexicalSelectionAlwaysOnDisplay";
<SelectionAlwaysOnDisplay  />
```

**Comment by msantic** [comment]
Hi,

First of all, thank you for this very useful plugin!

I’d like to ask if there’s a way to pass an onReposition function as an argument to markSelection. It seems markSelection already has some support for this, but in my use case I needed to apply a transform scale to the DOM node selection element since its parent was scaled.

Something like:

```ts
// Original code: https://github.com/facebook/lexical/blob/9028e4151c52d3957bf2df456655a9ee147ca5cb/packages/lexical-utils/src/selectionAlwaysOnDisplay.ts
function selectionAlwaysOnDisplay(editor: LexicalEditor, onReposition?: (domNodes: HTMLElement[]) => void): () => void {
  let removeSelectionMark: (() => void) | null = null;

  const onSelectionChange = () => {
    const domSelection = getSelection();
    const domAnchorNode = domSelection && domSelection.anchorNode;
    const editorRootElement = editor.getRootElement();

    const isSelectionInsideEditor =
      domAnchorNode !== null && editorRootElement !== null && editorRootElement.contains(domAnchorNode);

    if (isSelectionInsideEditor) {
      if (removeSelectionMark !== null) {
        removeSelectionMark();
        removeSelectionMark = null;
      }
    } else {
      if (removeSelectionMark === null) {
        removeSelectionMark = markSelection(editor, onReposition);
      }
    }
  };

  document.addEventListener('selectionchange', onSelectionChange);

  return () => {
    if (removeSelectionMark !== null) {
      removeSelectionMark();
    }
    document.removeEventListener('selectionchange', onSelectionChange);
  };
}

// Original code: https://github.com/facebook/lexical/blob/9028e4151c52d3957bf2df456655a9ee147ca5cb/packages/lexical-react/src/LexicalSelectionAlwaysOnDisplay.tsx#L13
export function SelectionAlwaysOnDisplay(onReposition?: (domNodes: HTMLElement[]) => void): null {
  const [editor] = useLexicalComposerContext();
  useEffect(() => {
    return selectionAlwaysOnDisplay(editor, onReposition);
  }, [editor]);

  return null;
}
```

Right now, instead of simply passing an onReposition argument, I’ve had to duplicate parts of the utils and the plugin source code from SelectionAlwaysOnDisplay.

Would it be possible to add direct support for this?

Thanks in advance!

**Reply by etrepum** [reply]
The best way to get this functionality sooner is to submit a PR
